### ISSUE TITLE
After installation of the plugin, QGIS must be restarted for the plugin to work

### ISSUE THREAD

**Issue by s-m-e** [qgist/workbench#5]

**Comment by AlisterH** [comment]
Note that if for any reason you unload and then restart the plugin it doesn't work until QGIS is restarted, either.

**Comment by s-m-e** [comment]
Indeed. Python does not have a proper concept for unloading modules. You can try to have them collected by the garbage collector, but this implies destroying all their references. The layer within QGIS that manages Python plugins is fragile and might keep unwanted references around, so the unloading and therefore reloading i.e. restarting of the plugin breaks.

The code in this repo needs a bit of an update. Happy to see users.
